api/devices/schema.py

```diff
@@ -123,6 +130,18 @@ def resolve_specific_device(self, info, device_id):
 
         return device
 
+    @Auth.user_roles('Admin')
+    def resolve_device_by_name(self, info, device_name):
+        query = Devices.get_query(info)
+        device_name = ''.join(device_name.split()).lower()
+        if not device_name:
+            raise GraphQLError("Please provide the device name")
+        for device in query:
+            exact_name = ''.join(device.name.split()).lower()
+            if exact_name == device_name:
```

Comment:
Is this filtering or searching?

Reply:
Am going through all the devices searching for a particular device name, so in a way, it's searching, but as @mnswaleh has pointed out I will now return  a list of devices with the same implementation

Reply:
@kwanj-k What is the difference between searching and filtering?

Reply:
If you use the filter,  you get a list with the particular devices, however, I would like to make sure there is no white space or difference in casing during comparison for every device.

Reply:
I understand what you meant now, I have made the changes and its now a search @Bonifase